include/zephyr/dt-bindings/reset/stm32f4_reset.h

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright (c) 2022 Google Inc
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_DT_BINDINGS_RESET_STM32F4_RESET_H_
+#define ZEPHYR_INCLUDE_DT_BINDINGS_RESET_STM32F4_RESET_H_
+
+#include "stm32-common.h"
+
+/* RCC bus reset register offset */
+#define STM32_RESET_BUS_AHB1 0x10
+#define STM32_RESET_BUS_AHB2 0x14
+#define STM32_RESET_BUS_AHB3 0x18
+#define STM32_RESET_BUS_APB1 0x20
+#define STM32_RESET_BUS_APB2 0x24
+
+/* AHB1 */
+#define STM32_RESET_GPIOA    STM32_RESET(AHB1, 0U)
+#define STM32_RESET_GPIOB    STM32_RESET(AHB1, 1U)
+#define STM32_RESET_GPIOC    STM32_RESET(AHB1, 2U)
+#define STM32_RESET_GPIOD    STM32_RESET(AHB1, 3U)
+#define STM32_RESET_GPIOE    STM32_RESET(AHB1, 4U)
+#define STM32_RESET_GPIOF    STM32_RESET(AHB1, 5U)
+#define STM32_RESET_GPIOG    STM32_RESET(AHB1, 6U)
+#define STM32_RESET_GPIOH    STM32_RESET(AHB1, 7U)
+#define STM32_RESET_GPIOI    STM32_RESET(AHB1, 8U)
+#define STM32_RESET_GPIOJ    STM32_RESET(AHB1, 9U)
+#define STM32_RESET_GPIOK    STM32_RESET(AHB1, 10U)
+#define STM32_RESET_CRC	     STM32_RESET(AHB1, 12U)
+#define STM32_RESET_DMA1     STM32_RESET(AHB1, 21U)
+#define STM32_RESET_DMA2     STM32_RESET(AHB1, 22U)
+#define STM32_RESET_DMA2D    STM32_RESET(AHB1, 23U)
+#define STM32_RESET_ETHMAC   STM32_RESET(AHB1, 25U)
+#define STM32_RESET_OTGHS    STM32_RESET(AHB1, 29U)
+#define STM32_RESET_RNG_AHB1 STM32_RESET(AHB1, 31U)
```

Comment:
Rather than providing this header, it would be easier to provide direct implementation in .dsti files:
```
			gpioa: gpio@40020000 {
				compatible = "st,stm32-gpio";
				gpio-controller;
				#gpio-cells = <2>;
				reg = <0x40020000 0x400>;
				clocks = <&rcc STM32_CLOCK_BUS_AHB1 0x00000001>;
				reset = <&rcc STM32_RESET(AHB1, 0U)>;
			};
```
which is more in line with what is done for clocks property and prevents the need to have SoC specific files (stm32f413_reset.h, stm32f446_reset.h).

Reply:
I took GD32 approach as example (eg. include/zephyr/dt-bindings/reset/gd32e10x.h). I will follow your suggestion and change it.

Reply:
Besides of this, does the code look good for you?

Reply:
Still looking into it, but this was the most visible issue I saw.
Otherwise the remaining part looks good, but I wanted to warn earlier on that part (I wish I could have warn before you perform the complete device tree completion).
Then, it would be nice to propose a specific PR for this whole development which is much boarder than "Improve multi-image support for STM32 UART and timer drivers"